addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NhcAction.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nikohomecontrol.internal.protocol;
+
+import org.openhab.binding.nikohomecontrol.handler.NikoHomeControlHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NhcAction} class represents the action Niko Home Control communication object. It contains all fields
+ * representing a Niko Home Control action and has methods to trigger the action in Niko Home Control and receive action
+ * updates.
+ *
+ * @author Mark Herwege
+ */
+public final class NhcAction {
```

Comment:
These classes exactly match the content of data elements in the Niko Home Control Controller. I wanted these to always stick to that content. As I need to access this class outside of the package, I had to make it public. Apart from that, there is no forcing reason for it. Is it better practice to not have them final?